cg/apps/loqus.py

```diff
@@ -25,58 +23,60 @@ class LoqusdbAPI():
     def __init__(self, config: dict):
         super(LoqusdbAPI, self).__init__()
 
-        self.loqusdb_config = config['loqusdb']['config_path']
-        self.loqusdb_binary = config['loqusdb']['binary_path']
-
-        # This will allways be the base of the loqusdb call
-        self.base_call = [self.loqusdb_binary, '--config', self.loqusdb_config]
-
-    def load(self, family_id: str, ped_path: str, vcf_path: str, vcf_sv_path: str,
-             gbcf_path: str) -> dict:
+        self.loqusdb_config = config["loqusdb"]["config_path"]
+        self.loqusdb_binary = config["loqusdb"]["binary_path"]
+        self.process = Process(self.loqusdb_binary, self.loqusdb_config)
+
+    def load(
+        self,
+        family_id: str,
+        ped_path: str,
+        vcf_path: str,
+        vcf_sv_path: str,
+        gbcf_path: str,
+    ) -> dict:
         """Add observations from a VCF."""
-        load_call = copy.deepcopy(self.base_call)
-        load_call.extend([
-            'load',
-            '-c', family_id,
-            '-f', ped_path,
-            '--variant-file', vcf_path,
-            '--sv-variants', vcf_sv_path,
-            '--check-profile', gbcf_path,
-            '--hard-threshold', '0.95',
-            '--soft-threshold', '0.90'
-            ])
+        load_call_parameters = [
+            "load",
+            "-c",
+            family_id,
+            "-f",
+            ped_path,
+            "--variant-file",
+            vcf_path,
+            "--sv-variants",
+            vcf_sv_path,
+            "--check-profile",
+            gbcf_path,
+            "--hard-threshold",
+            "0.95",
+            "--soft-threshold",
+            "0.90",
+        ]
 
         nr_variants = 0
+        self.process.run_command(load_call_parameters)
         # Execute command and print its stdout+stderr as it executes
```

Comment:
Now this will not be true right? This will wait for subprocess.run to finish, and then the stderr from the process can be checked.

Reply:
👍